backend/dataall/modules/shares_base/services/share_object_service.py

```diff
@@ -220,11 +227,13 @@ def submit_share_object(cls, uri: str):
         with context.db_engine.scoped_session() as session:
             share, dataset, states = cls._get_share_data(session, uri)
 
-            if not ShareObjectService.verify_principal_role(session, share):
-                raise PrincipalRoleNotFound(
-                    action='Submit Share Object',
-                    message=f'The principal role {share.principalRoleName} is not found.',
-                )
+            if share.principalType in [PrincipalType.ConsumptionRole.value, PrincipalType.Group.value]:
+                # TODO make it generic to non IAM role principals
+                if not ShareObjectService.verify_principal_role(session, share):
+                    raise PrincipalRoleNotFound(
+                        action='Submit Share Object',
+                        message=f'The principal role {share.principalRoleName} is not found.',
+                    )
```

Comment:
nit: move the line `if share.principalType in [PrincipalType.ConsumptionRole.value, PrincipalType.Group.value]:` and `return share.principalRoleName` if principalType is not one of the specified ones to verify

Instead of adding this condition to wrap around `if not ShareObjectService.verify_principal_role(session, share):` everywhere we call it

Reply:
I am redoing all this piece of code in the last PR of this series, so I will take it into account in that PR, it already looks completely different with the interface to inject validation of create share object